metadata-io/src/main/java/com/linkedin/metadata/graph/neo4j/Neo4jGraphService.java

```diff
@@ -94,6 +106,85 @@ public void addEdge(@Nonnull final Edge edge) {
     executeStatements(statements);
   }
 
+  @Nonnull
+  @Override
+  public EntityLineageResult getLineage(@Nonnull Urn entityUrn, @Nonnull LineageDirection direction,
+                                        GraphFilters graphFilters, int offset, int count, int maxHops) {
+    log.debug(String.format("Neo4j getLineage maxHops = %d", maxHops));
+
+    final String statement = generateLineageStatement(entityUrn, direction, graphFilters, maxHops);
+
+    List<Record> neo4jResult = statement != null ? runQuery(buildStatement(statement, new HashMap<>())).list() : new ArrayList<>();
+
+    neo4jResult = neo4jResult.stream()
+            .collect(Collectors.toMap(item -> item.values().get(2).asNode().get("urn").asString(),
+                    Function.identity(),
+                    (item1, item2) -> item1.get(1).size() < item2.get(1).size() ? item1 : item2))
+            .values()
+            .stream()
+            .collect(Collectors.toList());
+
+    LineageRelationshipArray relations = new LineageRelationshipArray();
+    neo4jResult.stream()
+            .skip(offset).limit(offset + count)
```

Comment:
Minor: are we sure that limit doesnt just accept count?

Reply:
You are right.